### ISSUE TITLE
Remove Prettier from eslint

### ISSUE THREAD

**Issue by turadg** [Agoric/Agoric#4339]
## What is the Problem Being Solved?
https://github.com/Agoric/agoric-sdk/pull/4330#discussion_r789387409

## Description of the Design

* Remove prettier concern from eslint
 * anyone relying on the eslint autofix can enable formatOnSave with https://github.com/Agoric/agoric-sdk/blob/master/scripts/configure-vscode.sh
* Add prettier check to CI
* `yarn lint-fix` in each package should still fix formatting (cc @kriskowal @warner )

## Security Considerations
n/a

## Test Plan

- [ ] prettier problems don't appear in IDE
- [ ] pretty-check failures fail CI

**Comment by dckc** [comment]
Note we rely on eslint for [@jessie-check](https://agoric.com/documentation/guides/js-programming/hardened-js.html#tool-support-eslint-config). Does this put that at risk? @michaelfig ?

**Comment by turadg** [comment]
@michaelfig pointed out that some people have `eslint --fix` set up to run on save, and taking Prettier out of that would stop their auto-formatting. [Prettier has editor integrations](https://prettier.io/docs/en/editors.html#visual-studio-code), but the change would impact developers.

So https://github.com/Agoric/agoric-sdk/pull/4342/ just downgrades the error to a warning, which is an improvement:
<img width="642" alt="Screen Shot 2022-01-21 at 3 28 24 PM" src="https://user-images.githubusercontent.com/21505/150616117-cdb7c6dc-ecef-4b64-a9af-c8c3ac671109.png">

TBD whether we want to remove it entirely.

**Comment by michaelfig** [comment]
> Note we rely on eslint for [@jessie-check](https://agoric.com/documentation/guides/js-programming/hardened-js.html#tool-support-eslint-config). Does this put that at risk?

No, we're good.  Eslint still runs from VSCode and CI, it's just making `eslint --fix` optional that we're talking about.

**Comment by erights** [comment]
This seems a relevant enough place to mention the anomalies seen at https://github.com/endojs/endo/pull/973#discussion_r790068936

I do not understand what I'm seeing there.

**Comment by mhofman** [comment]
Any reason people can't update their IDEs to run both eslint and prettier fix on save? VSCode should do prettier by default. I actually don't like having eslint fix on by default as that's usually more invasive changes, but I understand some may want to have it. The code hint overload can be real between all these tools.

Btw, do we know if the eslint-prettier plugin respects the `// prettier-ignore` instruction?

**Comment by dckc** [comment]
> Any reason people can't update their IDEs to run both eslint and prettier fix on save?

Clue us in as to how?

I have only the vaguest notion of how vs-code, eslint, and prettier interact.

**Comment by mhofman** [comment]
I have `"editor.defaultFormatter": "esbenp.prettier-vscode"` setup for a bunch of different filetypes in VSCode (JS, TS, html, json, etc)

Edit: Oh, and `"editor.formatOnSave": true`

**Comment by turadg** [comment]
> Clue us in as to how?

Made a script https://github.com/Agoric/agoric-sdk/blob/master/scripts/configure-vscode.sh

The settings that script spits out suppress eslint-prettier-plugin in VS Code so my main motivation for this PR is solved. I'll leave it open because the plugin still slows down linting unnecessarily but I'll move it to the Icebox for now.

**Comment by turadg** [comment]
https://github.com/Agoric/agoric-sdk/pull/4906#partial-pull-merging is failing on building the wallet-ui:
```
> @agoric/wallet-ui
$ yarn build:ses && yarn build:react
$ cp ../../../node_modules/ses/dist/lockdown.umd.js public/
$ react-scripts build
Creating an optimized production build...

Treating warnings as errors because process.env.CI = true.
Most CI servers set it automatically.

Failed to compile.

src/util/WalletBackendAdapter.js
  Line 25:6:  Replace `⏎···*·@param·{AsyncIterator<any[],·any[],·undefined>}·offersMembers⏎··` with `·@param·{AsyncIterator<any[],·any[],·undefined>}·offersMembers`  prettier/prettier
```
Build should not fail due to whitespace.

**Comment by mhofman** [comment]
Why is that file not ignored if it's a build output?

**Comment by turadg** [comment]
> Why is that file not ignored if it's a build output?

Not a build output. `src/util/WalletBackendAdapter.js` is a source file for the wallet-ui build. The build script fails if there are any warnings, even from ESLint ([ref](https://github.com/Agoric/agoric-sdk/pull/4859#issuecomment-1072443321)). Because Agoric's eslint config includes `prettier/prettier`, that the source didn't match the pretty format broke the build.

The reason it didn't match is that it automatically pulled in `prettier-plugin-jsdoc` because this build step doesn't have the `--no-plugin-search` that #4906 has. (I couldn't get `pluginSearchDirs: false` in the config to work.)

There are work-arounds but it's another example of a cost from making ESlint check Prettier style.

**Comment by turadg** [comment]
Just ran into something else that would improve by removing this: eslint warnings in Github UI:
<img width="1134" alt="Screen Shot 2022-04-01 at 8 06 47 AM" src="https://user-images.githubusercontent.com/21505/161290945-9ef57b6b-c90d-4306-9f28-dd597eac7744.png">

@FUDCo doesn't want any format-on-save and his draft PRs aren't meant to pass CI. The Prettier warnings clutter reading the code and provide no useful information.
